### ISSUE TITLE
Update the npm package

### ISSUE THREAD

**Issue by Wazbat** [smogon/pokemon-showdown#9428]
Hi there! I was trying to install the package locally on my machine, however I keep running into typescript compilation errors.

I asked around on the Typescript discord, and after investagion we found that the issue was due to this export in this file
```typescript
// node_modules/pokemon-showdown/.sim-dist/index.d.ts
export * from '../lib';
```
However this issue appears to have since been fixed in this repository, and just requires being published to npm

**Comment by mia-pi-git** [comment]
Hi!
We would advise using a `github:` dependency in your package.json instead of depending on the NPM package. 
(the NPM package is not currently maintained and is about two years out of date)

**Comment by Wazbat** [comment]
I did try that, however I run into issues both on windows and linux when installing that way
```json
{
  "dependencies": {
    "pokemon-showdown": "github:smogon/pokemon-showdown#7030687f1c09ff1f359448272affa098e57372e5"
  }
}
```
![image](https://user-images.githubusercontent.com/6253896/221993843-187b1650-a150-4695-940d-6cd54bc1a1c6.png)
![image](https://user-images.githubusercontent.com/6253896/221993876-db492b54-15b7-4cb0-83a4-afbdb06f8b91.png)

**Comment by monsanto** [comment]
Copying my response from another repo:

> I'm not opposed to releasing an NPM package of any of the repos in this organization, but it has to be zero overhead for our organization. So that probably means no meaningful semver (0.x?), and an automated build & upload process as part of GH actions. Or it could be a git branch on here that has the results of a build, which would save us the trouble of maintaining a different set of credentials. If someone wants to work on that then you have my blessing and thanks.

**Comment by Wazbat** [comment]
I could open a PR to configure github actions to deploy to github packages which would mean no additional credentials or anything. Could just use github releases for that too, so all that would be necessary is the creation of a release and it'd automatically build and upload

**Comment by monsanto** [comment]
Thanks for looking into this.

> I can do that if you're ok with occasionally creating releases in GH
I don't care that much what we use as long as it is convenient to use from `npm` and is fully automated. So if I have to sometimes go through the UI to create releases, then that's a no go. But if there's a separate timer making releases then that's OK too.

**Comment by Wazbat** [comment]
> So if I have to sometimes go through the UI to create releases, then that's a no go. But if there's a separate timer making releases then that's OK too.
Oh the idea I have for release creation would just be creating a release via the https://github.com/smogon/pokemon-showdown/releases page which is just a few steps `draft a new release -> increment version by 0.0.1 -> auto generate release notes -> create release`.

But even that can be automated with a scheduled github actions to just create a release on every commit... Though that wouldn't really be the best way (could just manually create a release once a month or something as it's only 4 clicks in the UI)

Regarding installing the package via `npm`, you can use github packages as I said before, however it appears that to actually install even a public package [you need to  create a personal GH token](https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#authenticating-to-github-packages), so it might be best to just publish to npm again, via this automation.

However to publish to NPM all that's needed is an NPM token which would be stored as an encrypted repository secret that noone can read, if you're ok with that I can open a PR with the neceserry CI to allow for automated publishing on releases (manual or scheduled)
![image](https://user-images.githubusercontent.com/6253896/222598060-b44356fb-2415-4853-8d72-6eef0a834acd.png)


### TLDR:
I can automate it all so there's no manual input, all I'd need is someone to create a repository secret with an NPM access token to publish to npm

**Comment by monsanto** [comment]
A PR publishing to NPM sounds good. I'll figure out the token

**Comment by Wazbat** [comment]
Ah perfect. I'll try to work on that soon then. And once you manage to create a token you can just create a repository secret for it. 
![image](https://user-images.githubusercontent.com/6253896/222984937-65ea7e8a-f74f-4a1d-bf4e-7daecc0a9fae.png)

**Comment by Stagory** [comment]
Hi all,
since there is no updated version of the npm package, is there a manual that describes how to install and use the current repository? If I'm trying to build the project I run into 1106 errors in 11 files.
`tsc`
![image](https://user-images.githubusercontent.com/83120682/225915226-46dc227f-198a-4e1f-813b-b22f0b1ea775.png)

**Comment by Zarel** [comment]
I tried building the latest version in the repository, and while it works like before, it still doesn't have TypeScript types.

**Comment by Stagory** [comment]
Okay, my mistake was that I had an outdated version of Node.js installed when I building the repository

**Comment by Wazbat** [comment]
> I tried building the latest version in the repository, and while it works like before, it still doesn't have TypeScript types.

I was under the impression that wasn't the case? Someone was trying to help me figure out why I was getting errors, and for them using a GitHub sha worked as it had since been fixed

**Comment by Zarel** [comment]
I suppose we'll find out when we release the next version.
